app/forms/teacher_reference_number_form.rb

```diff
@@ -0,0 +1,25 @@
+class TeacherReferenceNumberForm < Form
+  attribute :teacher_reference_number
+
+  validates :teacher_reference_number,
+    presence: {
+      message: ->(form, _) { form.i18n_errors_path("blank") }
+    }
+
+  validates :teacher_reference_number,
+    length: {
+      is: 7,
+      message: ->(form, _) { form.i18n_errors_path("length") }
+    }, if: -> { teacher_reference_number.present? }
+
+  def valid?(*)
+    self.teacher_reference_number = teacher_reference_number&.gsub(/\D/, "")
+    super
+  end
```

Comment:
Sure thing will add that and see what it looks like 👍

Reply:
Yeah I think that's nicer https://github.com/DFE-Digital/claim-additional-payments-for-teaching/pull/2705/commits/c32857dc81622e2338cbe382a94487c386c78357

Reply:
Will rebase this on @your 's changes once their on master to remove duplicating the before_valdiation stuff (and use the `ActiveModel::Validations::Callbacks` module rather than `ActiveModel::Callback`